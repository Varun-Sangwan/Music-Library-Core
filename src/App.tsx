import React from "react";
import { BrowserRouter as Router, Routes, Route } from "react-router-dom";
import { AuthProvider } from "./context/AuthContext";
import Login from "./pages/Login";
import Dashboard from "./pages/Dashboard";
import ProtectedRoute from "./components/ProtectedRoute";

const App = () => {
  return (
    <AuthProvider>
      <Router>
        <Routes>
          <Route path="/" element={<Login />} />
          <Route
            path="/dashboard"
            element={
              <ProtectedRoute>
                <Dashboard />
              </ProtectedRoute>
            }
          />
        </Routes>
      </Router>
    </AuthProvider>
  );
};

export default App;
