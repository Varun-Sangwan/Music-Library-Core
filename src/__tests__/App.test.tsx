import React from "react";
import { render } from "@testing-library/react";
import App from "../App";

test("renders App component correctly", () => {
  const { asFragment } = render(<App />);

  expect(asFragment()).toMatchSnapshot();
});
