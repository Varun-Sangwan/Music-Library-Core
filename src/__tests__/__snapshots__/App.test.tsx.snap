// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App component correctly 1`] = `
<DocumentFragment>
  <div
    class="login-container"
  >
    <div
      class="login-box"
    >
      <h2>
        Welcome!!
      </h2>
      <form>
        <div>
          <input
            placeholder="Enter username"
            required=""
            type="text"
            value=""
          />
        </div>
        <div>
          <input
            placeholder="Enter password"
            required=""
            type="password"
            value=""
          />
        </div>
        <button
          type="submit"
        >
          LOGIN
        </button>
      </form>
    </div>
  </div>
</DocumentFragment>
`;
